recorder/src/main/cpp/profile_writer.cc

```diff
@@ -182,3 +196,34 @@ void ProfileSerializingWriter::flush() {
     cpu_samples_flush_ctr = 0;
     w.flush();
 }
+
+#define METRIC_TYPE "profile_serializer"
+
+ProfileSerializingWriter::ProfileSerializingWriter(jvmtiEnv* _jvmti, ProfileWriter& _w, SiteResolver::MethodInfoResolver _fir, SiteResolver::LineNoResolver _lnr,
+                                                   PerfCtx::Registry& _reg, const SerializationFlushThresholds& _sft, const TruncationThresholds& _trunc_thresholds) :
+    jvmti(_jvmti), w(_w), fir(_fir), lnr(_lnr), reg(_reg), next_mthd_id(10), next_thd_id(3), next_ctx_id(5), sft(_sft), cpu_samples_flush_ctr(0), trunc_thresholds(_trunc_thresholds),
+
+    s_c_new_thd_info(GlobalCtx::metrics_registry->new_counter({METRICS_DOMAIN, METRIC_TYPE, "thd_rpt", "new"})),
+    s_c_new_ctx_info(GlobalCtx::metrics_registry->new_counter({METRICS_DOMAIN, METRIC_TYPE, "ctx_rpt", "new"})),
+    s_c_total_mthd_info(GlobalCtx::metrics_registry->new_counter({METRICS_DOMAIN, METRIC_TYPE, "mthd_rpt", "total"})),
+    s_c_new_mthd_info(GlobalCtx::metrics_registry->new_counter({METRICS_DOMAIN, METRIC_TYPE, "mthd_rpt", "new"})),
+
+    s_c_bad_lineno(GlobalCtx::metrics_registry->new_counter({METRICS_DOMAIN, METRIC_TYPE, "line_rpt", "bad"})),
+
+    s_c_frame_snipped(GlobalCtx::metrics_registry->new_counter({METRICS_DOMAIN, METRIC_TYPE, "backtrace_snipped"})),
+
+    s_m_stack_sample_err(GlobalCtx::metrics_registry->new_meter({METRICS_DOMAIN, METRIC_TYPE, "cpu_sample"}, "err")),
+    s_m_cpu_sample_add(GlobalCtx::metrics_registry->new_meter({METRICS_DOMAIN, METRIC_TYPE, "cpu_sample"}, "rpt")) {
+
+    s_c_new_mthd_info.clear();
```

Comment:
ProfileWriter is created and destroyed everytime we start and end work.

So the new_meter and new_histogram and other such functions do find_or_create internally.

The second time it works, it'll find existing values in that, so it clears them and starts over.

The other place this can be done is in the end, but that is dangerous, because it can kill it before we report the values.

May be I need a ttl feature :-), but I am thinking of leaving it the way it is for now, replace it with ttl-managed metrics later.